build/templates/E2E_Tests_On_Windows.yml

```diff
@@ -0,0 +1,145 @@
+      $EndToEndTestCommandToRunPart1 = '"' + "Run-Test -Exclude '$Part2' -Verbose *>&1 | Tee-Object $(System.DefaultWorkingDirectory)\artifacts\EndToEnd\FullLog_$(Build.BuildNumber).txt" +'"'
+      $EndToEndTestCommandToRunPart2 = '"' + "Run-Test -Exclude '$Part1' -Verbose *>&1 | Tee-Object $(System.DefaultWorkingDirectory)\artifacts\EndToEnd\FullLog_$(Build.BuildNumber).txt" +'"'
+      Write-Host "##vso[task.setvariable variable=EndToEndTestCommandToRunPart1]$EndToEndTestCommandToRunPart1"
+      Write-Host "##vso[task.setvariable variable=EndToEndTestCommandToRunPart2]$EndToEndTestCommandToRunPart2"
+
+- task: PowerShell@1
+  displayName: "Print Environment Variables"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      Write-Host "##vso[build.updatebuildnumber]$env:FullVstsBuildNumber"
+      Get-ChildItem Env: | Sort-Object Name | Format-Table -Wrap -AutoSize
+
+- task: DownloadBuildArtifacts@0
+  displayName: "Download Build artifacts"
+  inputs:
+    artifactName: "$(VsixPublishDir)"
+    downloadPath: "$(Build.Repository.LocalPath)/artifacts"
+
+- task: PowerShell@1
+  displayName: "Bootstrap.ps1"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)/scripts/e2etests/Bootstrap.ps1"
+    arguments: "-NuGetDropPath $(Build.Repository.LocalPath)\\artifacts\\$(VsixPublishDir) -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -verbose"
+
+- task: PowerShell@1
+  displayName: "SetupFunctionalTests.ps1"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\SetupFunctionalTests.ps1"
+
+- task: PowerShell@1
+  displayName: "SetupMachine.ps1"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\SetupMachine.ps1"
+
+- task: PowerShell@1
+  displayName: "InstallNuGetVSIX.ps1"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\InstallNuGetVSIX.ps1"
+    arguments: "-NuGetDropPath $(System.DefaultWorkingDirectory)\\artifacts\\$(VsixPublishDir) -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -NuGetVSIXID $(NuGetVsixId) -ProcessExitTimeoutInSeconds 180 -VSVersion 16.0"
+    failOnStandardError: "false"
+
+- task: PowerShell@1
+  displayName: "Collect VS Logs"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      Invoke-WebRequest -Uri $(CollectExeUrl) -OutFile $(System.DefaultWorkingDirectory)\\VSCollect.exe
+      $(System.DefaultWorkingDirectory)\\VSCollect.exe -zip:$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\e2e-collectlogs.zip
+  condition: "failed()"
+
+- ${{ if eq(parameters.Part, 'Part1') }}:
```

Comment:
as mentioned in the other comment, you're duplicating two long task definitions, just to have a single variable different. It would be better to have the test steps expect a single variable, and the contents of the variable is different, depending if it's step 1 or step 2.

Reply:
Thanks. Fixed.